src/patchkernel/patch_kernel_parallel.cpp

```diff
@@ -2947,6 +3234,54 @@ std::vector<int> PatchKernel::getNeighbourRanks()
 	return neighRanks;
 }
 
+/*!
+	Gets a constant reference to the ghost targets needed for data exchange.
+
+	\result A constant reference to the ghost targets needed for data
+	exchange.
+*/
+const std::unordered_map<int, std::vector<long>> & PatchKernel::getGhostVertexExchangeTargets() const
+{
+	return m_ghostVertexExchangeTargets;
+}
+
+/*!
+	Gets a constant reference to the ghost targets needed for data
+	exchange for the specified rank.
+
+	\param rank is the rank for which the information will be retreived
+	\result A constant reference to the ghost targets needed for data
+	exchange for the specified rank.
+*/
+const std::vector<long> & PatchKernel::getGhostVertexExchangeTargets(int rank) const
+{
+	return m_ghostVertexExchangeTargets.at(rank);
+}
+
+/*!
+	Gets a constant reference to the ghost sources needed for data exchange.
+
```

Comment:
The same as for target vertices. 
Here, we can also add a comment explaining that the sources are internal vertices and not ghost vertices.

> Sources ids are the ids of the local vertices, owned by the current rank, that are involved as sources in data exchange for each receving rank.